WeakAuras/Prototypes.lua

```diff
@@ -6354,21 +6354,31 @@ Private.event_prototypes = {
     name = L["Item Bonus Id Equipped"],
     statesParameter = "one",
     init = function(trigger)
+      trigger.itemBonusId = trigger.itemBonusId or ""
       local ret = [=[
-        local useLegendaryIcon = %s
-        local itemBonusId, itemId, itemName, itemIcon, itemSlot, itemSlotString = WeakAuras.GetBonusIdInfo(%q)
+        local fetchLegendaryPower = %s
+        local item = %q
+        local inverse = %s
+        local useItemSlot, slotSelected = %s, %d
+
+        local itemBonusId, itemId, itemName, icon, itemSlot, itemSlotString = WeakAuras.GetBonusIdInfo(item)
         local itemBonusId = tonumber(itemBonusId)
-        local icon = useLegendaryIcon and WeakAuras.GetLegendaryIcon(itemBonusId) or itemIcon
+        if fetchLegendaryPower then
+          itemName, icon = WeakAuras.GetLegendaryData(itemBonusId or item)
+        end
+
+        local slotValidation = ((useItemSlot and itemSlot == slotSelected) or (not useItemSlot))
```

Comment:
There's a extra () here compared to the style we usually use.

Reply:
Oops, that was left over from a lot of tinkering with inverse. I fix.